### PR TITLE
Approver Reports API

### PR DESCRIPTION
**Pull request by aditya999123**

### CODE REVIEW

File: src/components/schemas/report.yaml [hunk 1]
```diff
@@ -1,53 +1,94 @@
...
+      type: string
+      description: |
+        where was this report object created from
+    approvals:
+      type: array
+      items:
+        $ref: ./report_approval.yaml#/report_approval_out_embed
+      description: |
+        List of report_approval objects
+    num_expenses:
```
Comment (aditya999123) [inline src/components/schemas/report.yaml:90]: can add expense_embed_out later if required

File: src/components/schemas/report.yaml [hunk 2]
```diff
@@ -1,53 +1,107 @@
...
+        $ref: ./report_approval.yaml#/report_approval_out_embed
+      description: |
+        List of report_approval objects
+    expense_ids:
+      type: array
+      items:
+        $ref: ./fields.yaml#/id_string
+      description: |
+        Ids of expenses that are part of this report
+#    export_ids: # used in tpa_reports_view
```
Comment (aditya999123) [inline src/components/schemas/report.yaml:98]: check this

File: src/components/schemas/report.yaml [hunk 3]
```diff
@@ -1,53 +1,107 @@
...
+#      type: array
+#      items:
+#        $ref: ./fields.yaml#/id_string
+#      description: |
+#        id of expenses that are part of this report
+#    exported: # used in tpa_reports_view
```
Comment (aditya999123) [inline src/components/schemas/report.yaml:104]: check this

File: src/components/schemas/report.yaml [hunk 4]
```diff
@@ -1,53 +1,107 @@
...
+#      $ref: ./fields.yaml#/created_at
+#    is_manually_flagged:
+#      type: boolean
+#      description: |
+#        Determines whether this report was flagged manually
+#    is_policy_flaged:
+#      type: boolean
+#      description: |
+#        Determines whether this report was flagged from a policy
+    submitted_at:
```
Comment (aditya999123) [inline src/components/schemas/report.yaml:71]: this is required in slack

File: src/approver/paths/approver@reports_bulk_approve.yaml [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+  responses:
+    '200':
+      description: OK
+      content:
+        application/json:
+          schema:
+            type: object
+            required: []
+            additionalProperties: False
+            properties: {}
```
Comment (aditya999123) [inline src/approver/paths/approver@reports_bulk_approve.yaml:32]: returning nothing

File: src/approver/paths/approver@reports_bulk_approve.yaml [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+        example: 'updated_at.desc,id'
+  requestBody:
+    required: true
+    content:
+      application/json:
+        schema:
+          type: object
+          required: []
+          additionalProperties: False
+          properties: {}
```
Comment (aditya999123) [inline src/approver/paths/approver@reports_bulk_approve.yaml:31]: note empty body

File: src/approver/paths/approver@reports_bulk_approve.yaml [hunk 7]
```diff
@@ -0,0 +1,53 @@
+post:
+  tags:
+    - Reports
+  summary: Approve reports
+  operationId: approve-reports
+  parameters:
```
Comment (aditya999123) [inline src/approver/paths/approver@reports_bulk_approve.yaml:6]: note params in post call

File: src/approver/paths/approver@reports@{id}@approve.yaml [hunk 8]
```diff
@@ -0,0 +1,39 @@
+post:
+  tags:
+    - Reports
+  summary: Approve reports
+  operationId: approve-reports
```
Comment (jayachandhar) [inline src/approver/paths/approver@reports@{id}@approve.yaml:5]: parameters missing, 
```
parameters:
  in: query
  name: id
```

File: src/components/schemas/report.yaml [hunk 9]
```diff
@@ -1,53 +1,107 @@
...
+      $ref: ./fields.yaml#/currency
+    amount:
+      $ref: ./fields.yaml#/amount
+    tax:
+      $ref: ./fields.yaml#/amount
+    state:
+      type: string
+      enum:
+        - DRAFT
+        - PAYMENT_PROCESSING
```
Comment (jayachandhar) [inline src/components/schemas/report.yaml:31]: can we arrange this report processing order ?

**Pull request closed without merging**